TicTacToe2/Math/Vec2.cs

```diff
@@ -7,21 +7,86 @@ namespace TicTacToe2.Math
     class Vec2 // 2d vector
     {
         // Properties: X, Y (int). These properties should be read-only, meaning they should only have get, not set
+        public int X { get; }
+
+        public int Y { get; }
 
         // Constructor that takes x, y arguments
+        public Vec2(int x, int y)
+        {
+            X = x;
+            Y = y;
+        }
+
         // Constructor that takes no arguments and sets X and Y to zero
+        public Vec2()
+        {
+            X = 0;
+            Y = 0;
+        }
 
         // Methods to perform these operations:
         // 1. Add int value x to the X value of the vector
+        public Vec2 AddX(int x)
+        {
+            return new Vec2(X + x, Y);
+        }
+
         // 2. Add int value y to the Y value of the vector
+        public Vec2 AddY(int y)
+        {
+            return new Vec2(X, Y + y);
+        }
+
         // 3. Add int values x and y to the vector
+        public Vec2 AddXY(int x, int y)
+        {
+            return new Vec2(X + x, Y + y);
+        }
+
         // 4. Add another Vec2 to the vector
+        public Vec2 AddAnother(Vec2 One, Vec2 Two)
+        {
+            int newX = One.X + Two.X;
+            int newY = One.Y + Two.Y;
+            Vec2 Three = new Vec2(newX, newY);
+            return Three;
```

Comment:
This is not quite what I'm looking for. If you were to use this function, it would result in something like this:

```c#
var a = new Vec2(1, 1);
var b = new Vec2(2, 2);
var c = new Vec2(3, 3);
var d = a.AddAnother(b, c); // (5, 5) - a is ignored
```
I'm looking for a function that can be used like this:
```c#
var a = new Vec2(1, 1);
var b = new Vec2(2, 2);
var c = a.Add(b); // (3, 3)
```